generator/ServiceClientGeneratorLib/Generators/SourceFiles/ServiceConfig.tt

```diff
@@ -115,5 +116,26 @@ namespace <#=this.Config.Namespace#>
         public <#=parameter.nativeType#> <#=parameter.name#> { get; set; }
 
 <#}#>
+<# if (this.Config.EndpointsRuleSet != null) { #>
+        /// <summary>
+        /// Returns the endpoint that will be used for a particular request.
+        /// </summary>
+        /// <param name="request">Request for the desired service operation.</param>
+        /// <returns>The resolved endpoint for the given request.</returns>
+        public override Amazon.Runtime.Endpoints.Endpoint DetermineServiceOperationEndpoint(AmazonWebServiceRequest request)
+        {
+            var requestContext = new RequestContext(false, new <#=GeneratorHelpers.DetermineSigner(this.Config.ServiceModel.SignatureVersion, this.Config.ClassName)#>())
+            {
+                ClientConfig = this,
+                OriginalRequest = request,
+                Request = new DefaultRequest(request, ServiceId)
+            };
+
+            var executionContext = new Amazon.Runtime.Internal.ExecutionContext(requestContext, null);
+            var resolver = new Amazon<#=this.Config.ClassName#>EndpointResolver();
```

Comment:
Should we store the resolver as a member variable so we don't have to recreate for every request and possibly implement some caching in the resolver in the future.